### DISCUSSION TITLE
Loose files, and more GFF types?

### DISCUSSION THREAD

**Discussion by kanasc** [humhue/DAOTools#10] [Ideas]
This is a really handy tool; thank you for making it! I don't know how to export resources from the toolset in a way that doesn't include the English string, so I'd been laboriously editing them by hand; this is much more efficient (and I don't have to worry about human error). I have a couple of suggestions:

1. The ability to patch loose files in a folder, not just files inside an erf (handy for modders creating override mods rather than dazips)
2. Do you plan to expand this tool to patch other (v3.2) GFF types such as ARE, UTI, etc?

**Comment by humhue** [comment]
Thank you for reaching out to me.
The first issue would be pretty easy to address.
The second one is already implemented in a beta, unreleased version.
I wanted to add a GUI and to reduce the patching times by a lot, by also simplyifing the source, but ultimately I've never finished making these changes, because I thought they would've been of little help and not worth the hassle.
I guess I was wrong, at least partially.
The point is that it's likely, that this library is not being used much, maybe just by a handful of people.
At the end of the day, I made this to play the game in my native language with mods, but I've stopped playing it.
I don't know.
I might add these features, but it's not like it's going to benefit the community as I thought.
The code is pretty simple, and modifying it might be a good way to learn coding too.

How are you using or going to use this tool, if I may ask?

**Reply by kanasc** [reply]
I'm sorry to hear that this tool hasn't had as much traction as it deserves. And I understand stepping away from a project when you're not playing the game it was intended for any more. If you would like more interest and feedback, have you considered posting on Nexus? I suspect that most of the DAO modding community doesn't think to look on GitHub since the game is so old and nobody expects someone to make a new tool for it at this point. You could post your patched version of QUDAO as well, which I'm pretty sure would be in high demand - and would reach mod users as well as mod makers, and might then inspire interest in the tool that made it possible. If you do have the time and inclination to make a GUI, I think that would really help, because not everyone is comfortable with git or the command line.

I do suspect that a lot of the DAO community at this point is just accustomed to either having to play either in english or with bits of english scattered in if they use mods, or having to not use most mods - which is a shame, and might be part of why they aren't seeking out this tool, but it could change. The only thing I suppose this tool can't help with is mods that add things rather than edit vanilla things - since in order to be translatable, the mod would have to have its own talktable, and I don't think most do.

As for what I'm using the tool for, I'm currently editing some of the vanilla cutscenes in the toolset, which invariably export with english lines in them. Another project involves editing/exporting area files, which export with english names for map locations and the area itself in there - though I haven't actually tested if that overrides non-english text to english (which I really should do before I put work into it haha). So far I've been resetting them manually in pyGFF/tlkedit, but it's a slow process so this tool is very useful!

**Reply by humhue** [reply]
I'll think about it.
A problem of this tool is that, most mods are language independent, but a lot of them actually aren't.
QUDAO, for instance, might seem a technical patch, and well it is, kinda. But Qwinn made some weird little fixes, for example, he cut some lines from the dialogues because they were inconsistent with the game's plot. And that's probably ok, but he went on and chose to edit the voice overs and the models' lipsyncs too, in order to have more consistent dialogues, this is pretty crazy, fixes like these can't just be patched, they require a lot of manual work to be translated, or a tool to delete them from the patch.
Once, he edited a .plo file in the most naive way, all of the text was (erroneously) formatted in italics so he got the english file, removed the italics tags and packed it inside of the patch, forcing everyone with QUDAO installed to read that page in English, just because he didn't want to read that page formatted in italics.
Another problem is that some mods (like QUDAO), added new TLKStrings (I'm talking about in-game ones, not just credits) because he wanted to restore some of the content cut from the final game, and we can't just fix those lines automatically.

**Reply by humhue** [reply]
Hey @kanasc, I know it's been a while. I've released a new version of the tool with GUI and GFF 3.2 support. Also it lets you patch loose files but not a whole folder yet. It's a bit of a shortcoming I'll admit, but you can pack the folder into an erf, fix that and then unpack it. Thanks for this discussion :)
